packages/app/lib/ui/widgets/location_field.dart

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:app/ui/widgets/card.dart';
+import 'package:app/ui/widgets/card_action_button.dart';
+import 'package:app/ui/widgets/card_header.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:maplibre_gl/maplibre_gl.dart';
+
+typedef OnUpdate = void Function(
+    {required double latitude, required double longitude});
+
+class LocationField extends StatefulWidget {
+  final double latitude;
+  final double longitude;
+
+  final OnUpdate onUpdate;
+
+  const LocationField(
+      {super.key,
+      required this.longitude,
+      required this.latitude,
+      required this.onUpdate});
+
+  @override
+  State<LocationField> createState() => _LocationFieldState();
+}
+
+enum InputMode { read, edit }
+
+double DEFAULT_ZOOM = 10;
```

Comment:
I'm nod sure what the best default zoom is, maybe a little "further out" than it is now?